docs/architecture/env_isolation/README.md

```diff
@@ -0,0 +1,48 @@
+
+
+The purpose of this code is to ensure a stable environment for RP tasks,
+which ideally is (as) identical (as possible) to the environment the pilot
+is originally placed in.
+
+API / config changes
+
+   - a task has two sets of pre_exec directives:
+     - `pre_exec_env`: commands to prepare the environment
+       - eg., export, module load, source, ...
+       - runs on the agent node
+       - can be cached: identical command sets are run only once, env is then
+         reused for other tasks
+     - `pre_exec_cmd`: any action needed to run the task
+       - runs on the compute node
+       - will block all task resources apart from those assigned to rank 0
+       - runs in the task env (but not in the MPI env)
+     - backward compatibility: `pre_exec` is interpreted as `pre_exec_env`
+   - each LM has it's individual env settings (module load etc)
+     - any number of LMs can be configured for an agent
+     - TODO: clarify how a LM is selected for a task
```

Comment:
Will this replace the CU env parameter?

Reply:
No, it will augment it.  The goal is to avoid repeated `conda activate` calls for many small tasks at scale which kneels the file system, and also to make sure that the pilot env and modules do not creep into the task environments.

Reply:
Ah, I see. I would love an example, but do not let that keep the PR.